desktop/desktop
# [Demo] Stashing MVP

This PR ties together work from #6987 so that you can play with it in Desktop. This is a very rough demo, so when it does blow up, please comment here with what you did and maybe include a screenshot of chromeDev tools or your log file.


Steps to test:
1. Make some changes to some tracked files on some branch
2. Switch branches
3. Selected the first option (the 2nd can fail at the moment)
4. You will be taken to the branch you wanted to check out
5. Switch back to the branch you just left
6. Note the brutalist UI element notifying you of stashed changes

![2019-03-19 19 41 16](https://user-images.githubusercontent.com/1715082/54651166-9b668500-4a7f-11e9-8561-5d1da7a75b8f.gif)

![Screen Shot 2019-03-19 at 7 54 36 PM](https://user-images.githubusercontent.com/1715082/54651428-d7e6b080-4a80-11e9-906f-ac250d083fb5.png)

![2019-03-21 15 26 05](https://user-images.githubusercontent.com/1715082/54782949-21084300-4bee-11e9-9ddb-dca8a9547b46.gif)

**What's left to implement?**
1. Prompt if switching would overwrite the current stash
1. _Stashed Changes_ button is not in the _Changes_ view (doing so would require significant changes to our `List` component increasing complexity)
1. Clicking _Stashed Changes_ button does nothing (it will eventually present the option to apply/clear and will popup in the diff)
1. When the option to bring changes with you to new branch is selected, we need to immediately apply and clear the stash (this depends on #7036)
